lib/appium.js

```diff
@@ -60,6 +62,18 @@ class AppiumDriver extends BaseDriver {
     let d = new InnerDriver(this.args);
     let [innerSessionId, dCaps] = await d.createSession(caps, reqCaps, curSessions);
     this.sessions[innerSessionId] = d;
+
+    // Remove the session on unexpected shutdown, so that we are in a position
+    // to open another session later on.
+    // TODO: this should be removed and replaced by a onShutdown callback.
+    d.onUnexpectedShutdown
```

Comment:
We don't want to call `await this.sessions[sessionId].deleteSession();`. Btw I think long term it's a bad idea for AppiumDriver to extends BaseDriver , the sesson cache should be handled outside a driver.